src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/MetadataReader.netstandard.cs

```diff
@@ -13,13 +14,15 @@ public sealed partial class MetadataReader
         internal AssemblyName GetAssemblyName(StringHandle nameHandle, Version version, StringHandle cultureHandle, BlobHandle publicKeyOrTokenHandle, AssemblyHashAlgorithm assemblyHashAlgorithm, AssemblyFlags flags)
         {
             string name = GetString(nameHandle);
-            // compat: normalize 'null' culture name to "" to match AssemblyName.GetAssemblyName()
+            // compat: normalize Nil culture name to "" to match original behavior of AssemblyName.GetAssemblyName()
             string cultureName = (!cultureHandle.IsNil) ? GetString(cultureHandle) : "";
             var hashAlgorithm = (Configuration.Assemblies.AssemblyHashAlgorithm)assemblyHashAlgorithm;
-            byte[]? publicKeyOrToken = !publicKeyOrTokenHandle.IsNil ? GetBlobBytes(publicKeyOrTokenHandle) : null;
+            // compat: native BaseAssemblySpec::Init guarantees that publicKeyOrToken is never null.
```

Comment:
I would avoid referencing CoreCLR impl details here.

Reply:
These references get stale.

Reply:
I will just say - for compat reasons. 
I think this null->empty transformation correctly matches the native behavior, but in the future it will not matter, since managed implementation will become the baseline. 

I just wanted to add some comment, since what we are doing may not look right since it is equating null and empty, and the reason for that is basically backwards compat.